core/src/main/scala/org/apache/spark/resource/ResourceProfileManager.scala

```diff
@@ -73,11 +73,19 @@ private[spark] class ResourceProfileManager(sparkConf: SparkConf) extends Loggin
    * Gets the ResourceProfile associated with the id, if a profile doesn't exist
    * it returns the default ResourceProfile created from the application level configs.
    */
-  def resourceProfileFromId(rpId: Int): ResourceProfile = {
-    val rp = resourceProfileIdToResourceProfile.get(rpId)
-    if (rp == null) {
+  def resourceProfileFromId(rpId: Int): ResourceProfile = synchronized {
```

Comment:
Could you use `ReadWriteLock`? It seems that `addResourceProfile` is the only operation that needs `WriteLock`.

Reply:
yes that is a good idea, write should happen very infrequently, let me look more.